### PR TITLE
core: log parents[0] / node_id mismatch

### PR DESCRIPTION
**Pull request by spolu**
## Description

Introduce logs wherever we update or set parents in core to log when parents[0] and document_id mismatch.

Goal is to bring that log to 0

## Risk

N/A

## Deploy Plan

- deploy `core`

### CODE REVIEW

File: core/bin/core_api.rs [hunk 1]
```diff
@@ -1625,6 +1639,20 @@ async fn data_sources_documents_upsert(
...
+            parents_0 = match payload.parents.get(0) {
+                None => "None",
+                Some(p) => p,
+            },
+            node_type = "document",
+            operation = "upsert",
+            "[KWSEARCH] invariant_first_parent_self"
+        );
+    }
+
```
Comment (aubin-tchoi) [inline core/bin/core_api.rs:1655]: we should get rid of the one in `DataSource::upsert` no?

**Review by aubin-tchoi — APPROVED** [review]
LGTM 👍

File: core/bin/core_api.rs (hunk 1 above)
Comment (spolu) [reply core/bin/core_api.rs:1655]: did we add one?

Comment (spolu) [reply core/bin/core_api.rs:1655]: ah yes. removed.

**Pull request merged**